### ISSUE TITLE
MtxDateTimePicker does not update when type is changed.

### ISSUE THREAD

**Issue by pedroestabruxelles** [ng-matero/extensions#185]
Steps to reproduce from the example page:
- Pick a date/time
- Change type to Date

It gets stuck to the datetime type (2023-02-15 15:58), it should switch the view to date only (2023-02-15).

**Comment by HiteshWork13** [comment]
@pedroestabruxelles  which version of angular material you are using ?

**Comment by pranxy** [comment]
@nzbin  Some sort of feedback would be appreciated since this an actual bug and I don't understand why would not be fixed?

**Comment by wwaters-ms** [comment]
Yeah, I'm seeing the same issue with Angular 15. Setting  type="date" or type="datetime" works as expected, but setting [type]="type" displays datetime regardless of the value type has.

I basically had to write two separate components with the type hard coded and swap between them
